sqld/src/lib.rs

```diff
@@ -587,6 +591,54 @@ async fn run_storage_monitor(db_path: PathBuf, stats: Stats) -> anyhow::Result<(
     Ok(())
 }
 
+async fn run_checkpoint_cron(db_path: PathBuf, period: Duration) -> anyhow::Result<()> {
+    const RETRY_INTERVAL: Duration = Duration::from_secs(60);
+    let data_path = db_path.join("data");
+    tracing::info!("setting checkpoint interval to {:?}", period);
+    let mut interval = interval(period);
```

Comment:
The default "missed tick behavior" for Tokio interval is to aggresively catch up, ticking multiple times: https://docs.rs/tokio/latest/tokio/time/enum.MissedTickBehavior.html . That makes little sense with checkpointing, and it can be a performance issue, since we'd repeatedly take a write lock multiple times in a row. Judging from the docs, perhaps https://docs.rs/tokio/latest/tokio/time/enum.MissedTickBehavior.html#variant.Delay is a better choice?